### PR TITLE
Update doc preview tutorial to all static sites

### PR DESCRIPTION
**Pull request by tuliren**
The current tutorial for doc preview actually applies to all static sites. To add Stoat as an option to a bunch of static site deployment documentations, we need a tutorial page. Modifying the doc preview tutorial seems the easiest.

This is needed by https://github.com/stoat-dev/vite/pull/1.

### CODE REVIEW

**Review by jrhizor — APPROVED** [review]

File: docs/docs/concepts/static-hosting.mdx [hunk 1]
```diff
@@ -8,7 +8,7 @@ We have tutorials to help you set up static hosting for various types of build o
 * [Build Reports (tests, coverage, etc.)](../tutorials/view-reports)
 * [Images](../tutorials/preview-images)
 * [Videos](../tutorials/preview-videos)
-* [Static Documentation (like Swagger, Javadocs, etc.)](../tutorials/preview-docs)
+* [Static Site (like Swagger, Javadocs, etc.)](../tutorials/preview-static-site)
```
Comment (jrhizor) [inline docs/docs/concepts/static-hosting.mdx:11]:
```suggestion
* [Static Sites (like Swagger, Javadocs, etc.)](../tutorials/preview-static-site)
```

nit: all the others are plural

File: docs/docs/partials/_tutorial_list.mdx [hunk 2]
```diff
@@ -1,7 +1,7 @@
 * [Create and update a pull request comment](/docs/tutorials/create-and-update-comment)
 * [View build time graph in every pull request](/docs/tutorials/build-runtimes)
 * [View reports (tests, coverage, etc) in pull requests](/docs/tutorials/view-reports)
-* [Preview documentation in pull requests](/docs/tutorials/preview-docs)
+* [Preview static site in pull requests](/docs/tutorials/preview-static-site)
```
Comment (jrhizor) [inline docs/docs/partials/_tutorial_list.mdx:4]:
```suggestion
* [Preview static sites in pull requests](/docs/tutorials/preview-static-site)
```

File: docs/docs/tutorials/preview-static-site.mdx [hunk 3]
```diff
@@ -1,17 +1,18 @@
 ---
 sidebar_position: 106
-title: Docs Previews
+title: Static Site Previews
 ---
 
-# Preview documentation in PRs
+# Preview Static Site in PRs
```
Comment (jrhizor) [inline docs/docs/tutorials/preview-static-site.mdx:6]:
```suggestion
# Preview Static Sites in PRs
```

File: docs/docs/why-stoat/javascript.mdx [hunk 4]
```diff
@@ -40,7 +40,7 @@ We also have tutorials for specific use-cases:
 * [View build time graph in every pull request](../tutorials/build-runtimes)
 * [Preview generated images in pull requests](../tutorials/preview-images)
 * [Preview generated videos in pull requests](../tutorials/preview-videos)
-* [Preview documentation in pull requests](../tutorials/preview-docs)
+* [Preview static site in pull requests](../tutorials/preview-static-site)
```
Comment (jrhizor) [inline docs/docs/why-stoat/javascript.mdx:43]:
```suggestion
* [Preview static sites in pull requests](../tutorials/preview-static-site)
```

File: docs/docs/why-stoat/java.mdx [hunk 5]
```diff
@@ -40,7 +40,7 @@ HTML test reports using Gradle and can be used as a reference.
 
 To get started with Stoat as a Java engineer, try out the following tutorials:
 * [View reports (tests, coverage, etc) in pull requests](../tutorials/view-reports)
-* [Preview documentation in pull requests](../tutorials/preview-docs)
+* [Preview static site in pull requests](../tutorials/preview-static-site)
```
Comment (jrhizor) [inline docs/docs/why-stoat/java.mdx:43]:
```suggestion
* [Preview static sites in pull requests](../tutorials/preview-static-site)
```

**Comment by jrhizor** [comment]
After going back through the docs, it doesn't seem like there's a difference between "view build reports" and "static site previews". I think the distinction is really that if someone comes to that page and has build reports or javadocs or swagger docs, they probably aren't thinking that they have a static site, so it might not be obvious that "static site" is where they should go. 

Maybe we should be splitting these into many more sections that are more "by use case"?
- Test results
- Test coverage
- Static Analysis
- API Docs
- Code Docs
- Static Sites

The actual guides would be nearly identical, and we could have customized expected `path` values for each, so hopefully it wouldn't be much harder to maintain since we can use a shared component.

What do you think?

**Comment by tuliren** [comment]
Sure, as long as these docs are generated from shared components, we can have separate docs for each application.

**Pull request merged**